go/pkg/bertyprotocol/store_metadata_index.go

```diff
@@ -167,12 +168,39 @@ func (m *metadataStoreIndex) handleGroupAddDeviceSecret(event proto.Message) err
 	}
 
 	if m.ownMemberDevice.device.Equals(senderPK) {
-		m.sentSecrets[string(e.DestMemberPK)] = struct{}{}
+		m.sentSecrets[string(e.DevicePK)+string(e.DestMemberPK)] = struct{}{}
```

Comment:
not sure if there is already another mutex in a top-level, but this part may be problematic if run concurrently
in general, when managing maps in async environments, always use mutex around the map manipulation

Reply:
yes handled functions are already protected by a mutex

Reply:
you need to be sure that 100% of the codebase manipulating the map use mutex when manipulating the map